lib/routes/token.js

```diff
@@ -236,6 +240,32 @@ module.exports = {
       }
       return vals;
     })
+    .then(function(vals) {
+      // Incremental authorization.
+      if (params.existing_grant) {
+        return db.getRefreshToken(encrypt.hash(params.existing_grant)).then(function (tok) {
+          if (!tok) {
+            throw new AppError.unauthorized('Unknown existing grant');
+          }
+          if (tok.userId.toString('hex') !== vals.userId.toString('hex') ||
+              tok.clientId.toString('hex') !== vals.clientId.toString('hex')) {
+            throw new AppError.unauthorized('Invalid grant');
+          }
+          vals.scope = Scope(vals.scope).merge(Scope(tok.scope)).values();
+          return vals;
+        });
+      } else if (vals.includeGrantedScopes) {
```

Comment:
What, if anything, should happen if both `existing_grant` and `include_granted_scopes` are specified at the same time?

Reply:
From the [proposed spec](https://tools.ietf.org/html/draft-ietf-oauth-incremental-authz-00#section-4) it looks like `include_granted_scopes` is only for confidential clients whereas `existing_grant` is intended primarily for public clients but can be used for confidential clients (at least that's my interpretation of `Public clients (and confidential clients using this technique)`. It does not state what should happen if both are specified for a confidential client, my guess would be the union but I also don't understand how the scopes granted using `existing_grant` would be anything but a subset of scopes granted in `include_granted_scopes`.

Asking for clarification for this scenario from the spec authors might be useful since they are actively seeking feedback at this time.

Reply:
Ping @WilliamDenniss, do you know what should happen in this case?